spark/src/main/scala/org/apache/spark/sql/delta/ColumnWithDefaultExprUtils.scala

```diff
@@ -117,6 +119,29 @@ object ColumnWithDefaultExprUtils extends DeltaLogging {
     val constraints = mutable.ArrayBuffer[Constraint]()
     // Column names for which we will track high water marks.
     val track = mutable.Set[String]()
+    val generatedColumnsValidateMode =
+      GeneratedColumnValidateOnWriteMode.fromConf(data.sparkSession.sessionState.conf)
+    generatedColumnsValidateMode match {
+      case GeneratedColumnValidateOnWriteMode.LOG_ONLY |
+           GeneratedColumnValidateOnWriteMode.ASSERT =>
+        try {
+          GeneratedColumn.validateGeneratedColumns(data.sparkSession, schema)
+        } catch {
+          case NonFatal(e) =>
+            recordDeltaEvent(
+              deltaLog,
+              "delta.generatedColumns.writeValidationFailure",
+              data = Map(
+                "errorClassName" -> e.getClass,
```

Comment:
Actually: you may want to log the `errorClass` instead here when available for Spark exception, otherwise you'll get a very generic `AnalyisisException`